### PR TITLE
DAOS-6486 dfuse: Switch back to using directories for UNS entries.

### PR DESCRIPTION
**Pull request by ashleypittman**
Change from using FIFOs back to directories for backing UNS entries.

This does not work with directory metadata caching enabled, so turn
off caching for directories.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
Ready to land, please review.

**Review by johannlombardi — APPROVED** [review]

File: src/client/dfuse/ops/lookup.c [hunk 1]
```diff
@@ -23,19 +23,12 @@ dfuse_reply_entry(struct dfuse_projection_info *fs_handle,
...
-	 */
-
-	if (S_ISFIFO(ie->ie_stat.st_mode)) {
-		if (!is_new) {
-			ie->ie_stat.st_mode &= ~S_IFIFO;
-			ie->ie_stat.st_mode |= S_IFDIR;
-		}
-	} else {
-		entry.attr_timeout = ie->ie_dfs->dfs_attr_timeout;
+	/* Do not cache directly attributes as this does not work with uns */
```
Comment (daltonbohning) [inline src/client/dfuse/ops/lookup.c:26]: Minor typo:
```suggestion
	/* Do not cache directory attributes as this does not work with uns */
```

**Review by daltonbohning — APPROVED** [review]
LGTM.
Could you please remind me why we're switching back to directories instead of FIFOs?

File: src/client/dfuse/ops/lookup.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfuse/ops/lookup.c:26]: I'm keen to move forward so I'll address this in a future PR.

**Comment by mchaarawi** [comment]
@ashleypittman i don't see here where you revert the change you made at the DFS level.
please push a PR to remove the FIFO handling in DFS

**Comment by ashleypittman** [comment]
> @ashleypittman i don't see here where you revert the change you made at the DFS level.
> please push a PR to remove the FIFO handling in DFS

Will do.  I considered the dfs changes, they do work (in fact it would be possible to create a FIFO now through dfuse and dfs) so I wasn't sure if we wanted these removed or not.

Would you prefer the entire support to be removed or just the create part?

**Comment by mchaarawi** [comment]
> > @ashleypittman i don't see here where you revert the change you made at the DFS level.
> > please push a PR to remove the FIFO handling in DFS
> 
> Will do. I considered the dfs changes, they do work (in fact it would be possible to create a FIFO now through dfuse and dfs) so I wasn't sure if we wanted these removed or not.
> 
> Would you prefer the entire support to be removed or just the create part?

considering we do not claim we have it supported and we don't have testing anywhere, i'd rather we just remove the entire support for now. it's minimal effort anyway if we want to add it later.

**Pull request merged**